### PR TITLE
[MAINT] minor fixes

### PR DESCRIPTION
**Pull request by enver-bisevac**
- Set data to repository only if new version of object
- Sort rules by priority
- Added test case

### CODE REVIEW

File: __tests__/usecases/rules_priority.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
+{
```
Comment (knagurski) [inline __tests__/usecases/rules_priority.json:1]: Not necessary, but usually a good idea to run things like this through prettier for consistency

Comment (enver-bisevac) [reply __tests__/usecases/rules_priority.json:1]: Done

File: src/repository.ts [hunk 2]
```diff
@@ -59,38 +65,48 @@ export class StorageRepository implements Repository {
...
         this.cache.set(segmentKey, segment);
       }
       return segment;
     }
     return undefined;
   }
 
+  private checkFlagVersion(key: string, flag: FeatureConfig): boolean {
+    const oldFlag = this.getFlag(key, false); // dont set cacheable, we are just checking the version
+    return !oldFlag || !oldFlag.version || oldFlag.version < flag?.version;
```
Comment (knagurski) [inline src/repository.ts:102]: You _could_ simplify, but it's fine as-is
```suggestion
    return !oldFlag?.version || oldFlag.version < flag?.version;
```

File: src/evaluator.ts [hunk 3]
```diff
@@ -216,6 +216,11 @@ export class Evaluator {
     if (!target || !rules) {
       return undefined;
     }
+
+    rules.sort((a: ServingRule, b: ServingRule) =>
+      a.priority > b.priority ? 1 : b.priority > a.priority ? -1 : 0,
```
[the remark this answers is not in the document]
Comment (enver-bisevac) [reply src/evaluator.ts]: Make sense

File: src/repository.ts [hunk 4]
```diff
@@ -59,38 +65,48 @@ export class StorageRepository implements Repository {
...
-      if (segment) {
+      if (segment && cacheable) {
         this.cache.set(segmentKey, segment);
       }
       return segment;
     }
     return undefined;
   }
 
+  private checkFlagVersion(key: string, flag: FeatureConfig): boolean {
```
Comment (knagurski) [inline src/repository.ts:100]: Just personal preference, but I prefer a the function name to describe what it'll give me back. In the case of a boolean like this, something like `isFlagOutdated`

File: src/repository.ts (hunk 2 above)
Comment (enver-bisevac) [reply src/repository.ts:102]: make sense :)

**Review by knagurski — APPROVED** [review]
I don't like this.


I love it! ❤️

**Pull request merged**